users/init.sls

```diff
@@ -78,9 +78,12 @@ users_{{ name }}_user:
     {% if 'uid' in user -%}
     - uid: {{ user['uid'] }}
     {% endif -%}
-    {% if 'password' in user -%}
+    {% if 'password in user -%}
     - password: '{{ user['password'] }}'
     {% endif -%}
+    {% if empty_password is defined -%}
```

Comment:
``empty_password`` isn't passed into the template as a context variable, it's a key of ``user`` thus line 84 would never evaluate to true.
To use ``'empty_password' in user`` is the right way to test for a key's existence.